frontend/test/federations/federations.store.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { createTestStore } from '../test.helpers';
+import { containerMockFactory, prepareMockViews } from '../containers/containers.fixtures';
+
+describe('Federations: store', () => {
+	let dispatch; let
+		getState;
+	const projectId = 'projectId';
+
+	beforeEach(() => {
+		({ dispatch, getState } = createTestStore());
+		dispatch(ProjectsActions.setCurrentProject(projectId));
+	});
+
+	const createAndAddFederationToStore = (federationOverrides = {}) => {
+		const mockFederation = federationMockFactory(federationOverrides);
+		dispatch(FederationsActions.fetchFederationsSuccess(projectId, [mockFederation]));
+		return mockFederation;
+	};
+
+	it('should create a federation successfully', () => {
+		createAndAddFederationToStore();
+		const mockFederation = federationMockFactory();
+		dispatch(FederationsActions.createFederationSuccess(projectId, prepareMockNewFederation(mockFederation), mockFederation._id));
+		const federationIsIncluded = selectFederationById(getState(), mockFederation._id);
+		expect(federationIsIncluded).toBeTruthy();
+	});
+
+	it('should fetch federations successfully', () => {
+		const mockFederations = times(5, () => federationMockFactory());
+		dispatch(FederationsActions.fetchFederationsSuccess(projectId, mockFederations));
+		const federations = selectFederations(getState());
+		expect(federations).toEqual(mockFederations);
+	});
+
+	it('should set a federation as favourite', () => {
+		const mockFederation = createAndAddFederationToStore({ isFavourite: false });
+		let favouriteFederations = selectFavouriteFederations(getState());
+		expect(favouriteFederations).toEqual([]);
+		const { _id: federationId, isFavourite } = mockFederation;
+		dispatch(FederationsActions.setFavouriteSuccess(projectId, federationId, !isFavourite));
+		favouriteFederations = selectFavouriteFederations(getState());
+		expect(favouriteFederations).toEqual([{ ...mockFederation, isFavourite: true }]);
+	});
+
+	it('should fetch a federation\'s stats', () => {
+		const mockFederation = createAndAddFederationToStore();
+		const stats = federationMockStats();
+		dispatch(FederationsActions.fetchFederationStatsSuccess(projectId, mockFederation._id, stats));
+		const federationFromState = selectFederationById(getState(), mockFederation._id);
+		expect(federationFromState.code).toEqual(stats.code);
+		expect(federationFromState.lastUpdated).toEqual(new Date(stats.lastUpdated));
+		expect(federationFromState.status).toEqual(stats.status);
+		expect(federationFromState.category).toEqual(stats.category);
+	});
+
+	it('should fetch a federation\'s views', () => {
+		const { _id: federationId } = createAndAddFederationToStore();
+		const mockViews = prepareMockViews();
+		dispatch(FederationsActions.fetchFederationViewsSuccess(projectId, federationId, mockViews));
+		const viewsFromState = selectFederationById(getState(), federationId).views;
+		expect(viewsFromState).toBe(mockViews);
```

Comment:
According to the jest docs: `Use .toBe to compare primitive values or to check referential identity`, The tests still pass, but I think it's better to use `toEqual` which does a deep comparison when it comes to objects